[18.0][MIG] account_invoice_fiscal_position_update: Migration to 18.0

rov-adhoc opened the pull request.

alexis-via committed:
Rename to account_invoice_fiscal_position_update

mourad-ehm committed:
[ADD] add tests

rousseldenis committed:
[10.0][FIX] account_invoice_fiscal_position_update: On invoice creation, tax mapping could be wrong set

RoelAdriaans committed:
[MIG] account_invoice_fiscal_position_update: Migration to 11.0

pedrobaeza committed:
[FIX+IMP] account_invoice_fiscal_position_update: Fix acc code + README by fragments

Commit:
Translated using Weblate (Dutch)

Currently translated at 100,0% (4 of 4 strings)

Translation: account-invoicing-11.0/account-invoicing-11.0-account_invoice_fiscal_position_update
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-11-0/account-invoicing-11-0-account_invoice_fiscal_position_update/nl_NL/

rschnapka committed:
Translated using Weblate (German)

Currently translated at 100,0% (4 of 4 strings)

Translation: account-invoicing-11.0/account-invoicing-11.0-account_invoice_fiscal_position_update
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-11-0/account-invoicing-11-0-account_invoice_fiscal_position_update/de/

bjeficent committed:
[MIG] account_invoice_fiscal_position_update: Migration to version 12.0

alexis-via committed:
account_invoice_fiscal_position_update: use method get_invoice_line_account()

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (4 of 4 strings)

Translation: account-invoicing-12.0/account-invoicing-12.0-account_invoice_fiscal_position_update
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-12-0/account-invoicing-12-0-account_invoice_fiscal_position_update/de/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (4 of 4 strings)

Translation: account-invoicing-12.0/account-invoicing-12.0-account_invoice_fiscal_position_update
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-12-0/account-invoicing-12-0-account_invoice_fiscal_position_update/pt_BR/

MiquelRForgeFlow committed:
[FIX] *: reference_type doesn't exist anymore

MiquelRForgeFlow committed:
[IMP] account_invoice_fiscal_position_update: black, isort

MiquelRForgeFlow committed:
[MIG] account_invoice_fiscal_position_update: Migration to 13.0

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 100.0% (4 of 4 strings)

Translation: account-invoicing-13.0/account-invoicing-13.0-account_invoice_fiscal_position_update
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-13-0/account-invoicing-13-0-account_invoice_fiscal_position_update/nl/

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (4 of 4 strings)

Translation: account-invoicing-13.0/account-invoicing-13.0-account_invoice_fiscal_position_update
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-13-0/account-invoicing-13-0-account_invoice_fiscal_position_update/es/

nicomacr committed:
[13.0] [FIX] account_invoice_fiscal_position_update: recompute the taxes after changed

nicomacr committed:
[13.0] [FIX] account_invoice_fiscal_position_update: Filter only lines without comments or notes.

nicomacr committed:
[FIX] account_invoice_fiscal_position_update: Mark lines to recompute taxes when this was changed.

Otherwise the taxes are not changed when change the fpos.

alexis-via committed:
[MIG] account_invoice_fiscal_position_update: Migration to 14.0

nicomacr committed:
[MIG] account_invoice_fiscal_position_update: Migration to 15.0

willytal committed:
[FIX] account_invoice_fiscal_position_update: fix tests

TT32113

MiquelRForgeFlow committed:
[FIX] account_invoice_fiscal_position_update: avoid error in tests

Don't create new variant, just use the default one.

ljsalvatierra-factorlibre committed:
[MIG] account_invoice_fiscal_position_update: Migration to 16.0

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_fiscal_position_update
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_fiscal_position_update/hr/

Gelo-fl committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_fiscal_position_update
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_fiscal_position_update/es/

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (6 of 6 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_fiscal_position_update
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_fiscal_position_update/pt_BR/

moitabenfdz committed:
[IMP] account_invoice_fiscal_position_update: pre-commit stuff

moitabenfdz committed:
[17.0][MIG] account_invoice_fiscal_position_update

moitabenfdz committed:
[FIX]account_invoice_fiscal_position_update: fix error when changing fiscal position with a line without product and description

rov-adhoc committed:
[MIG] account_invoice_fiscal_position_update: Migration to 18.0

gonzalolenzi commented:
LGTM

rov-adhoc commented:
@pedrobaeza Hi! I would like to ask you if it's something missing in this migration in order to be approve
Thanks in advance!

pedrobaeza commented:
A PSC or maintainer should review it: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#review

legalsylvain commented:
Hi @rov-adhoc thanks for porting this module. could you rebase, to re build runboat. I'd like to know if the bug described here is still present in the 18.0 branch.  (https://github.com/OCA/account-invoicing/issues/1896)
Thanks !

rov-adhoc commented:
@legalsylvain thanks for your message. I was able to reproduce the same error in the 18.0 branch. Do you know if anyone is currently working on a fix for this issue?

legalsylvain commented:
> Do you know if anyone is currently working on a fix for this issue?

I don't know. I just raised this error, using the V16 module. Let's wait what maintainers and original developpers are saying about that bug.

StefanRijnhart committed:
[FIX] account_invoice_fiscal_position_update: fix tax incl. to tax excl.

Fixes #1896

This change causes the subtotal amount to remain the same when a fiscal position
change triggers a tax update from tax included in price to tax excluded from
price.

Tests are added for conversion between tax excluded to tax included as well, but
in this case the behaviour is not 'fixed'. The standard behaviour of Odoo in
case of tax excl. to tax incl. is to adjust the subtotal amount. The same
behaviour is present in the function that this module provides. It might not
be what users expect, but it seems that Odoo does not support to preserve the
subtotal amount in this case.

StefanRijnhart reviewed (commented):
Please consider https://github.com/adhoc-dev/oca-account-invoicing/pull/1

rov-adhoc commented:
@StefanRijnhart HI! I just included your PR in this migration! thanks

StefanRijnhart commented:
Thanks! Can you drop the merge commit?

rov-adhoc commented:
> Thanks! Can you drop the merge commit?

Done

StefanRijnhart commented:
Yes, but now you squashed my changes into the migration commit. That is not correct. The original commit should be kept.

rov-adhoc commented:
> Yes, but now you squashed my changes into the migration commit. That is not correct. The original commit should be kept.

Hope now its ok

StefanRijnhart reviewed (approved):
Yes, thank you!

legalsylvain commented:
/ocabot merge nobump

StefanRijnhart commented:
/ocabot merge nobump

StefanRijnhart commented:
/ocabot migration account_invoice_fiscal_position_update

pedrobaeza commented:
This module is not needed anymore since 17.0: https://github.com/OCA/account-invoicing/pull/1935#issuecomment-2704413797

I think we should remove it. @StefanRijnhart @rov-adhoc @cav-adhoc

StefanRijnhart commented:
@pedrobaeza Maybe it's a bit harsh to remove it, but maybe we can remove the code, set it to uninstallable and just note the obsolescense in the module description? This would not break automatic deploys where the module is already installed I think.

pedrobaeza commented:
OK, that seems the best path for those with the module installed. @rov-adhoc are you able to do it?

rov-adhoc commented:
@pedrobaeza @StefanRijnhart Hey! I’ve created the PR for this. Let me know if any changes are needed! 
PR -->
https://github.com/OCA/account-invoicing/pull/1941
